cloudinit/distros/__init__.py

```diff
@@ -241,7 +257,11 @@ def apply_network_config(self, netconfig, bring_up=False) -> bool:
         """
         # This method is preferred to apply_network which only takes
         # a much less complete network config format (interfaces(5)).
-        network_state = parse_net_config_data(netconfig)
+        if self._is_netplan() and netconfig.get("version") == 2:
```

Comment:
I don't like how the network rendering abstraction is leaking into the distro code here. Do you think it's possible to push this section of code down into either `parse_net_config_data` or even into `NetworkStateInterpreter`?

Reply:
Logic moved to `NetworkStateInterpreter`. The downside is that we have to pass the renderer around, but it looks more cohesive in this way. Thanks for the comment.